### PR TITLE
Add torch.nan and torch.inf in docs

### LINKED ISSUE #131040: Create the pages which explain `torch.nan` and `torch.inf` in the doc
### 📚 The doc issue

I can use `torch.nan` and `torch.inf` as shown below but I cannot find the pages which explain `torch.nan` and `torch.inf` in [the doc](https://pytorch.org/docs/stable/index.html):

```python
import torch

torch.nan, torch.inf
# (nan, inf)
```

### Suggest a potential alternative/fix

So, the pages which explain `torch.nan` and `torch.inf` should be created in the doc.

cc @albanD

### PR DESCRIPTION
**Pull request by ashwanirathee**
Fixes #131040

This is my first pr here, please guide me if I get something wrong.

### CODE REVIEW

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -2597,6 +2597,16 @@ def add_docstr_all(method, docstr):
 """,
 )
 
+add_docstr_all(
```
Comment (albanD) [inline torch/_tensor_docs.py:2600]: These don't exist on Tensor right?

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -5380,6 +5380,22 @@ def merge_dicts(*dicts):
     ),
 )
 
+add_docstr(
```
Comment (albanD) [inline torch/_torch_docs.py:5383]: nit: we might want to add the same for e and pi given https://github.com/pytorch/pytorch/blob/0246b28510f11364d0e1fb7cc4eb353bd649a8f5/torch/__init__.py#L1681

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by ashwani-rathee** [comment]
I looked into the issue, method we are using here i.e. add_docstr alias for -> _add_docstr uses a decorator which populates __doc__ of functions, classes but given none of torch.e, torch.pi has __doc__, it fails the test. How do you suggest to proceed further?

**Pull request closed without merging**